### PR TITLE
Update to WCSLIB 7.11

### PR DESCRIPTION
**Pull request by mcara**
### Description
Update bundled WCSLIB to version ~7.10~ (skipped) **7.11** which includes a bug fix related to swapping -TAB axes.

Closes https://github.com/astropy/astropy/pull/13124

EDIT: Fix spacetelescope/jdaviz#1272

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by dhomeier** [comment]
I suppose that `tabcpy` bug never showed up in the existing tests. If so, should one be added?

File: docs/changes/wcs/13171.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Updated bundled WCSLIB to version 7.10 with a bug fix in ``tabcpy()`` and other enhancements. For a full list of bug fixes see https://www.atnf.csiro.au/people/mcalabre/WCS/CHANGES
```
Comment (pllim) [inline docs/changes/wcs/13171.feature.rst:1]: I don't think you can use "feature" for a bugfix backport to LTS. @saimn can confirm.

Comment (mcara) [reply docs/changes/wcs/13171.feature.rst:1]: OK. I'll fix that

Comment (mcara) [reply docs/changes/wcs/13171.feature.rst]: done

**Review by pllim — APPROVED** [review]
This actually fixes spacetelescope/jdaviz#1272 , so you get approval from me. Thanks! 😸

**Comment by mcara** [comment]
> I suppose that `tabcpy` bug never showed up in the existing tests. If so, should one be added?

I am not sure what the bug was based on the CHANGELOG description. But what prompted this fix was that swapped -TAB axis was behaving differently from the original WCS. https://github.com/astropy/astropy/pull/13124 should cover the swapping issue.

**Comment by mcara** [comment]
I ported unit tests from https://github.com/astropy/astropy/pull/13124 to this PR and added check of round-tripping

**Comment by saimn** [comment]
It's 7.10 right, not 7.11 ?

**Comment by mcara** [comment]
7.11 was released a few minutes ago. I am working on replacing 7.10->7.11 in this PR (so basically skip 7.10)

**Comment by olebole** [comment]
@pllim I only learned from this PR that there are new versions of wcslib, and I am quite busy today. I could only make a review tonight (UTC/CEST) or so.

**Comment by mcara** [comment]
@pllim Try https://github.com/spacetelescope/jdaviz/issues/1272 again

**Comment by dhomeier** [comment]
Still wishing for a GH tool that would show you all changes other than version no./date. 😆

**Review by dhomeier — APPROVED** [review]
Ran the tests against system-wcs 7.7 and 7.9-7.11 and they do as they should, thanks.
Just waiting for @pllim to report on the jdaviz issues.

**Comment by pllim** [comment]
> mcara deleted a comment from pep8speaks

Hey, I saw that! 😸

**Comment by pllim** [comment]
@meeseeksdev backport to v5.0.x

**Comment by pllim** [comment]
@meeseeksdev backport to v5.1.x

**Pull request merged**